src/gpuarray_reduction.c

```diff
@@ -12,115 +12,712 @@
+		case GA_LONG16:
+		case GA_LONG:
+		  *property = "LONG_MIN";
+		break;
+		case GA_LONGLONG:
+		  *property = "LLONG_MIN";
+		break;
+		case GA_BOOL:
+		case GA_UBYTE2:
+		case GA_UBYTE3:
+		case GA_UBYTE4:
+		case GA_UBYTE8:
+		case GA_UBYTE16:
+		case GA_UBYTE:
+		case GA_USHORT2:
+		case GA_USHORT3:
+		case GA_USHORT4:
+		case GA_USHORT8:
+		case GA_USHORT16:
+		case GA_USHORT:
+		case GA_UINT2:
+		case GA_UINT3:
+		case GA_UINT4:
+		case GA_UINT8:
+		case GA_UINT16:
+		case GA_UINT:
+		case GA_ULONG2:
+		case GA_ULONG3:
+		case GA_ULONG4:
+		case GA_ULONG8:
+		case GA_ULONG16:
+		case GA_ULONG:
+		case GA_ULONGLONG:
+		case GA_SIZE:
+		  *property = "0";
+		break;
+		case GA_HALF:
+		case GA_FLOAT:
+		case GA_DOUBLE:
+		case GA_QUAD:
+		  *property = "NAN";
+		break;
+		default:
+		  return GA_UNSUPPORTED_ERROR;
+	}
+
+	return GA_NO_ERROR;
+}
+
+/**
+ * @brief Get an expression representing a suitable initialization value for
+ *        the given datatype and a min-reduction operation.
+ *
+ * @param [in]  typecode  Typecode of the type whose initializer is to be
+ *                        requested.
+ * @param [out] property  A pointer to a string. On return it will be set to
+ *                        the initializer expression.
+ * @return Zero if successful; Non-zero if the datatype is not supported.
+ */
+
+static int   reduxGetMaxInit               (int typecode, const char** property){
+	switch (typecode){
+		case GA_BOOL:
+		  *property = "1";
+		break;
+		case GA_BYTE2:
+		case GA_BYTE3:
+		case GA_BYTE4:
+		case GA_BYTE8:
+		case GA_BYTE16:
+		case GA_BYTE:
+		  *property = "SCHAR_MAX";
+		break;
+		case GA_UBYTE2:
+		case GA_UBYTE3:
+		case GA_UBYTE4:
+		case GA_UBYTE8:
+		case GA_UBYTE16:
+		case GA_UBYTE:
+		  *property = "UCHAR_MAX";
+		break;
+		case GA_SHORT2:
+		case GA_SHORT3:
+		case GA_SHORT4:
+		case GA_SHORT8:
+		case GA_SHORT16:
+		case GA_SHORT:
+		  *property = "SHRT_MAX";
+		break;
+		case GA_USHORT2:
+		case GA_USHORT3:
+		case GA_USHORT4:
+		case GA_USHORT8:
+		case GA_USHORT16:
+		case GA_USHORT:
+		  *property = "USHRT_MAX";
+		break;
+		case GA_INT2:
+		case GA_INT3:
+		case GA_INT4:
+		case GA_INT8:
+		case GA_INT16:
+		case GA_INT:
+		  *property = "INT_MAX";
+		break;
+		case GA_UINT2:
+		case GA_UINT3:
+		case GA_UINT4:
+		case GA_UINT8:
+		case GA_UINT16:
+		case GA_UINT:
+		  *property = "UINT_MAX";
+		break;
+		case GA_LONG2:
+		case GA_LONG3:
+		case GA_LONG4:
+		case GA_LONG8:
+		case GA_LONG16:
+		case GA_LONG:
+		  *property = "LONG_MAX";
+		break;
+		case GA_ULONG2:
+		case GA_ULONG3:
+		case GA_ULONG4:
+		case GA_ULONG8:
+		case GA_ULONG16:
+		case GA_ULONG:
+		  *property = "ULONG_MAX";
+		break;
+		case GA_LONGLONG:
+		  *property = "LLONG_MAX";
+		break;
+		case GA_ULONGLONG:
+		  *property = "ULLONG_MAX";
+		break;
+		case GA_HALF:
+		case GA_FLOAT:
+		case GA_DOUBLE:
+		case GA_QUAD:
+		  *property = "NAN";
+		break;
+		default:
+		  return GA_UNSUPPORTED_ERROR;
+	}
+
+	return GA_NO_ERROR;
```

Comment:
I think that an additional property on the types for min/max values would be better than this.